### PR TITLE
Bugfix: Lr finder and hparams compatibility

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/1983
Solves compatibility problem between the (new) hparams interface and the learning rate finder

### CODE REVIEW

File: pytorch_lightning/utilities/parsing.py [hunk 1]
```diff
@@ -140,3 +140,57 @@ def __repr__(self):
         rows = [tmp_name.format(f'"{n}":', self[n]) for n in sorted(self.keys())]
         out = '\n'.join(rows)
         return out
+
+
+def lightning_hasattr(model, attribute):
```
Comment (Borda) [inline pytorch_lightning/utilities/parsing.py:145]: mind add test / doctests

**Comment by awaelchli** [comment]
If setting the attribute failed before, it means that we have no tests for the LRfinder that involves setting the learning rate. I would have expected such a test case with this fix. What do you think?

**Comment by SkafteNicki** [comment]
@awaelchli I am pretty sure that there should be test for this, but I will definitely check up on it and add if missing (fix if broken)

**Comment by SkafteNicki** [comment]
@awaelchli you were right, the test did not check this since the base testing model `model=EvalModelTemplate()` both have an `model.learning_rate` attribute and a `model.hparams.learning_rate` attribute with the latter never being checked. I have changed this now.

File: tests/utilities/parsing.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        hparams = TestHparamsNamespace()
+    model2 = TestModel2()
+
+    class TestModel3:  # test for hparams dict
+        hparams = TestHparamsDict
+    model3 = TestModel3()
+
+    class TestModel4:  # fail case
+        batch_size = 1
+    model4 = TestModel4()
```
Comment (Borda) [inline tests/utilities/parsing.py:26]:
```suggestion
    model1 = TestModel1()

    class TestModel2:  # test for hparams namespace
        hparams = TestHparamsNamespace()

    model2 = TestModel2()

    class TestModel3:  # test for hparams dict
        hparams = TestHparamsDict

    model3 = TestModel3()

    class TestModel4:  # fail case
        batch_size = 1

    model4 = TestModel4()
```

**Comment by williamFalcon** [comment]
looks like GPU tests are failing?
@awaelchli, @SkafteNicki

**Comment by awaelchli** [comment]
@williamFalcon these failing tests come from master

**Comment by Borda** [comment]
if has been rebased (master is fixed) but it is still failing, @SkafteNicki mind check it?
